printers/printerTree.php

```diff
@@ -135,36 +137,50 @@ private function str_contains(string $haystack, string $needle){
         return strpos($haystack, $needle) !== false;
     }
 
-    private function _fillTree($tree, $keys, $item, $parentId) {
+    private function _fillTree($tree, $keys, $item, $parentId, $myNs) {
         if (empty($keys)){ // We've reach the end of the journey. We register the data of $item
             if($item['type'] === 'd') {
                 $tree->self = $item;
             } else {
-                $tree->pages []= $item;
+                if ($myNs == $item['id']) {
+                    $tree->self = $item;
+                } else {
+                    if (!isset($tree->children[$item['id']])) {
+                        if ('d' !== $item['type']) {
```

Comment:
The answer is simple, I don't know if there is other type than d or f, I write in that manner because it shout work for everything other than d, I worried if I wrote "f" I lose some node.